pyaedt/edb_core/edb_data/siwave_simulation_setup_data.py

```diff
@@ -0,0 +1,897 @@
+from pyaedt.edb_core.edb_data.hfss_simulation_setup_data import EdbFrequencySweep
+from pyaedt.edb_core.general import convert_netdict_to_pydict
+from pyaedt.edb_core.general import convert_pydict_to_netdict
+from pyaedt.generic.general_methods import generate_unique_name
+from pyaedt.generic.general_methods import pyaedt_function_handler
+
+
+class SiwaveAdvancedSettings(object):
+    def __init__(self, parent):
+        self._parent = parent
+
+    @property
+    def sim_setup_info(self):
+        """Edb Internal Simulation Setup Object."""
+        return self._parent._edb_sim_setup_info
+
+    @property
+    def include_inter_plane_coupling(self):
+        """Get/Set the InterPlane Coupling.
+        The setter will enable custom settings.
+
+        Returns
+        -------
+        bool
+        """
+        return self.sim_setup_info.SimulationSettings.AdvancedSettings.IncludeInterPlaneCoupling
+
+    @property
+    def xtalk_threshold(self):
+        """Get/Set the XTalk Threshold.
+        The setter will enable custom settings.
+
+        Returns
+        -------
+        str
+        """
+        return self.sim_setup_info.SimulationSettings.AdvancedSettings.XtalkThreshold
+
+    @property
+    def min_void_area(self):
+        """Get/Set the minimum void area to be included.
+
+        Returns
+        -------
+        str
+        """
+        return self.sim_setup_info.SimulationSettings.AdvancedSettings.MinVoidArea
+
+    @property
+    def min_pad_area_to_mesh(self):
+        """Get/Set the minimum void pad are to mesh to be included.
+
+        Returns
+        -------
+        str
+        """
+        return self.sim_setup_info.SimulationSettings.AdvancedSettings.MinPadAreaToMesh
+
+    @property
+    def min_plane_area_to_mesh(self):
+        """Get/Set the minimum plane area to mesh be included.
+
+        Returns
+        -------
+        str
+        """
+        return self.sim_setup_info.SimulationSettings.AdvancedSettings.MinPlaneAreaToMesh
+
+    @property
+    def snap_length_threshold(self):
+        """Get/Set the snapping length threshold.
+
+        Returns
+        -------
+        str
+        """
+        return self.sim_setup_info.SimulationSettings.AdvancedSettings.SnapLengthThreshold
+
+    @property
+    def return_current_distribution(self):
+        """Get/Set the Flag to enable return current distribution.
+        Accurately model the change of the characteristic impedance of transmission lines caused by
+        a discontinuous ground plane.
+        Instead of injecting the return current of a trace into a single point on the ground plane,
+        the return current for a high impedance trace is now spread out.
+        The trace return current is not distributed when all traces attached to a node
+        have a characteristic impedance less than 75 ohms, or if the difference between
+        two connected traces is less than 25 ohms.
```

Comment:
```suggestion
        """Whether to enable the return current distribution.
        This property is used to accurately model the change of the characteristic impedance
        of transmission lines caused by a discontinuous ground plane. Instead of injecting
        the return current of a trace into a single point on the ground plane,
        the return current for a high impedance trace is spread out.
        The trace return current is not distributed when all traces attached to a node
        have a characteristic impedance less than 75 ohms or if the difference between
        two connected traces is less than 25 ohms.
```